internal/controller/nginx/config/validation/generic.go

```diff
@@ -106,3 +110,75 @@ func (GenericValidator) ValidateEndpoint(endpoint string) error {
 
 	return nil
 }
+
+const (
+	variableNameFmt    = `\$[a-z_]+`
+	variableNameErrMsg = "must start with '$' followed by lowercase letters and underscores only"
+)
+
+var variableNameRegexp = regexp.MustCompile("^" + variableNameFmt + "$")
+
+// ValidateNginxVariableName validates an nginx variable name.
+func (GenericValidator) ValidateNginxVariableName(name string) error {
+	if !variableNameRegexp.MatchString(name) {
+		examples := []string{
+			"$upstream_addr",
+			"$remote_addr",
+		}
+
+		return errors.New(k8svalidation.RegexError(variableNameFmt, variableNameErrMsg, examples...))
+	}
+
+	return nil
+}
+
+var (
+	PlusAllowedLBMethods = map[ngfAPI.LoadBalancingType]struct{}{
+		ngfAPI.LoadBalancingTypeRoundRobin:                 {},
+		ngfAPI.LoadBalancingTypeLeastConnection:            {},
+		ngfAPI.LoadBalancingTypeIPHash:                     {},
+		ngfAPI.LoadBalancingTypeRandom:                     {},
+		ngfAPI.LoadBalancingTypeHash:                       {},
+		ngfAPI.LoadBalancingTypeHashConsistent:             {},
+		ngfAPI.LoadBalancingTypeRandomTwo:                  {},
+		ngfAPI.LoadBalancingTypeRandomTwoLeastConnection:   {},
+		ngfAPI.LoadBalancingTypeLeastTimeHeader:            {},
+		ngfAPI.LoadBalancingTypeLeastTimeLastByte:          {},
+		ngfAPI.LoadBalancingTypeLeastTimeHeaderInflight:    {},
+		ngfAPI.LoadBalancingTypeLeastTimeLastByteInflight:  {},
+		ngfAPI.LoadBalancingTypeRandomTwoLeastTimeHeader:   {},
+		ngfAPI.LoadBalancingTypeRandomTwoLeastTimeLastByte: {},
+	}
+
+	OSSAllowedLBMethods = map[ngfAPI.LoadBalancingType]struct{}{
+		ngfAPI.LoadBalancingTypeRoundRobin:               {},
+		ngfAPI.LoadBalancingTypeLeastConnection:          {},
+		ngfAPI.LoadBalancingTypeIPHash:                   {},
+		ngfAPI.LoadBalancingTypeRandom:                   {},
+		ngfAPI.LoadBalancingTypeHash:                     {},
+		ngfAPI.LoadBalancingTypeHashConsistent:           {},
+		ngfAPI.LoadBalancingTypeRandomTwo:                {},
+		ngfAPI.LoadBalancingTypeRandomTwoLeastConnection: {},
+	}
+)
+
+func (GenericValidator) ValidateLoadBalancingMethod(method string, plusEnabled bool) error {
```

Comment:
does this function need to live on the GenericValidator? Couldn't we just call it in the upstream settings Validate function?